container_opts.go

```diff
@@ -316,7 +316,10 @@ func WithNewSpec(opts ...oci.SpecOpts) NewContainerOpts {
 			return err
 		}
 		c.Spec, err = typeurl.MarshalAny(s)
-		return err
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
Returning error directly should be cleaner and not introduce any bugs.
And when `err ! = nil` there is no handling of err.

Reply:
I see, I looked at it carefully, there is no problem, except for the look and feel